core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -442,60 +444,39 @@ sealed abstract class Doc extends Product with Serializable {
     loop(Left(this) :: Nil, Doc.empty)
   }
 
-  /**
-   * This method is similar to flatten, but returns None if no
-   * flattening was needed (i.e. if no newlines or unions were present).
-   *
-   * As with flatten, the resulting Doc (if any) will never render any
-   * newlines, no matter what width is used.
-   */
-  def flattenOption: Option[Doc] = {
-    val res = flattenBoolean
-    if (res._2) Some(res._1) else None
-  }
+  private[paiges] def tryFlatten: Doc = flatten getOrElse this
 
-  /**
-   * Convert this Doc to a single-line representation.
-   *
-   * All newlines are replaced with spaces (and optional indentation
-   * is ignored). The resulting Doc will never render any newlines, no
-   * matter what width is used.
-   */
-  def flatten: Doc = flattenOption getOrElse this
-
-  // return the flattened doc, and if it is different
-  private def flattenBoolean: (Doc, Boolean) = {
```

Comment:
previously we were using that flattenBoolean simplifies the Doc also (it makes sure we have the correct associativity on Concat, and it removes needless Align, Nest, etc...

Reply:
Oh, thanks for pointing that out.  I didn't appreciate it until now.  Will consider how to get it back.